packages/taquito-sapling/src/error.ts

```diff
@@ -30,3 +30,38 @@ export class TreeConstructionFailure extends Error {
     super(message);
   }
 }
+
+/**
+ *  @category Error
+ *  @description Error indicating that the memo is invalid
+ */
+export class InvalidMemo extends Error {
+  public name = 'InvalidMemo';
+  constructor(memo: string, errorDetail: string) {
+    super(`The memo '${memo}' is invalid. ${errorDetail}`);
+  }
+}
+
+/**
+ *  @category Error
+ *  @description Error indicating that there is not enough balance to prepare the sapling transaction
+ */
+export class InsufficientBalance extends Error {
+  public name = 'InsufficientBalance';
+  constructor(realBalance: string, amountToSpend: string) {
+    super(
+      `The transaction would spend ${amountToSpend} mutez while the balance is only ${realBalance} mutez.`
```

Comment:
I think the message wording can be a bit ambiguous in a sense that it _might_ still spend mutez. Maybe we could consider changing it to something like `Unable to spend ${amountToSpend} mutez while balance is only ${realBalance} mutez`

Reply:
Thanks, I will make the change.